services/outputhost/messagecache.go

```diff
@@ -401,7 +403,7 @@ func (msgCache *cgMsgCache) utilHandleRedeliveryTicker() {
 		msgCache.reinjectlastAckMsg() // injects the last acked message if it is not already injected
 	}
 
-	for _, cache := range timerCaches {
+	for i, cache := range timerCaches {
```

Comment:
if i am reading it right .. it seems like this function (utilHandleRedeliveryTicker) gets called once in 500ms (redeliveryInterval). for our nack-redelivery case, it seems like if the message is not ready for redelivery .. it will be checked again only after 500ms, even though our nack-redelivery delay is just 25ms.